test/torch/tensors/test_autograd.py

```diff
@@ -0,0 +1,149 @@
+import torch
+import syft
+
+from syft.frameworks.torch.tensors.interpreters import AutogradTensor
+
+
+def test_wrap():
+    """
+    Test the .on() wrap functionality for AutogradTensor
+    """
+
+    x_tensor = torch.Tensor([1, 2, 3])
+    x = AutogradTensor().on(x_tensor)
+
+    assert isinstance(x, torch.Tensor)
+    assert isinstance(x.child, AutogradTensor)
+    assert isinstance(x.child.child, torch.Tensor)
+
+
+def test_add_backwards(workers):
+    
+    alice = workers['alice']
+    a = torch.tensor([3, 2., 0], requires_grad=True)
+    b = torch.tensor([1, 2., 3], requires_grad=True)
+
+    a = a.send(alice, local_autograd=True)
+    b = b.send(alice, local_autograd=True)
+
+    a_torch = torch.tensor([3, 2.0, 0], requires_grad=True)
+    b_torch = torch.tensor([1, 2.0, 3], requires_grad=True)
+
+    c = a + b
+    c_torch = a_torch + b_torch
+
+    c.backward(torch.ones(c.shape).send(alice))
+    c_torch.backward(torch.ones(c_torch.shape))
+
+    assert (a.grad.get() == a_torch.grad).all()
+    assert (b.grad.get() == b_torch.grad).all()
+
+
+def test_mul_backwards(workers):
+    
+    alice = workers['alice']
+    a = torch.tensor([3, 2., 0], requires_grad=True)
+    b = torch.tensor([1, 2., 3], requires_grad=True)
+
+    a = a.send(alice, local_autograd=True)
+    b = b.send(alice, local_autograd=True)
+
+    a_torch = torch.tensor([3, 2.0, 0], requires_grad=True)
+    b_torch = torch.tensor([1, 2.0, 3], requires_grad=True)
+
+    c = a * b
+    c_torch = a_torch * b_torch
+
+    c.backward(torch.ones(c.shape).send(alice))
+    c_torch.backward(torch.ones(c_torch.shape))
+
+    assert (a.grad.get() == a_torch.grad).all()
+    assert (b.grad.get() == b_torch.grad).all()
+    
+    a.get();
+    b.get();
+    c.get();
+
+
+def test_sqrt_backwards(workers):
```

Comment:
test_sqrt_backwards fails with:
E   TypeError: can not serialize 'AutogradTensor' object
msgpack/_packer.pyx:279: TypeError